LuhnAlgorithm/src/main/java/com/exercism/luhn/Mod10Validator.java

```diff
@@ -0,0 +1,84 @@
+package com.exercism.luhn;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Mod10Validator {
+    private static final String FILE_NAME = "TestData.txt";
+    private ArrayList<String> listToValidate;
+    Scanner scanner;
+
+    public Mod10Validator() {
+        try {
+            ClassLoader classLoader = getClass().getClassLoader();
+            File file = new File(classLoader.getResource(FILE_NAME).getFile());
```

Comment:
File reading should be an util function in a separate class. Organize fields, variables and methods in such a way, so that they are decoupled. `Mod10Validator` class should have a method like `validate(String)` and\or `validate(List<String>)` that returns true or false. All the the other functionality should be extracted in separate class(/es).